### DISCUSSION TITLE
Switching to new key after accidental key revocation

### DISCUSSION THREAD

**Discussion by danielhatton** [zeapo/Android-Password-Store#2323] [Q&A]
I've been using Password Store happily for some time (thanks all developers!), with the password store in a local directory, and the OpenKeyChain  authentication mode.  Unfortunately, I had a little accident in OpenKeyChain and revoked the key I'd been using for Password Store.  As a result, I can't add any new passwords to the store (although I can still access the passwords that were already in there).  I've created a new key in OpenKeychain (and added it to the authorized_keys file of the local SimpleSSHD ssh server), but I can't find anything in the Password Store documentation about how to get Password Store to migrate to using the new key.  Please advise.

**Comment by danielhatton** [comment]
Thanks for the conversion, @msfjarvis.  Good idea.

**Comment by msfjarvis** [comment]
Migrating keys in pass requires a lot of decrypt/encrypt cycles that are very slow and cumbersome to execute with OpenKeychain so APS doesn't support the functionality, at least for now. You can however still perform the migration through the pass CLI on a computer. I can enclose instructions here if you're able to share what OS you have secure access to, since this will involve transferring your store to said computer.

**Comment by danielhatton** [comment]
On Fri, 6 Jan 2023, Harsh Shandilya wrote:

> Migrating keys in pass requires a lot of decrypt/encrypt cycles that
> are very slow and cumbersome to execute with OpenKeychain so APS
> doesn't support the functionality, at least for now. You can however
> still perform the migration through the pass CLI on a computer. I
> can enclose instructions here if you're able to share what OS you
> have secure access to, since this will involve transferring your
> store to said computer.

Thanks.  I've got an Ubuntu box from which I can ssh/sftp in to the
phone.
